### PR TITLE
Deploy RMQ 3.8.12 by default

### LINKED ISSUE #595: Bump default RabbitMQ version
**Is your feature request related to a problem? Please describe.**
We should keep up to date with RabbitMQ releases. Version 3.8.11 addressed a good number of bugs related to quorum queues and it added [individual health checks via management API](https://github.com/rabbitmq/rabbitmq-management/issues/844). 

**Describe the solution you'd like**
RabbitMQ 3.8.12 is producing release candidates. We should bump RabbitMQ to 3.8.12 shortly after it gets released.

**Additional context**
Our system tests use a not so great aliveness check via the management API. As part of bumping to 3.8.12, we should change the system tests to use an individual health check e.g. `/api/health/checks/protocol-listener/amqp091`

### PR DESCRIPTION
**Pull request by ChunyiLyu**
This closes #595 

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes

Deploy RMQ 3.8.12 by default.

## Additional Context

Should we add a warning in our README to warn people about the maintenance mode data loss issue for mirror queues?

## Local Testing

All passing.

### CODE REVIEW

**Review by mkuratczyk — APPROVED** [review]
Failing kubectl plugin tests seem to be a flake. We may want to improve them but they seem to be a false negative as far as this change is concerned.

File: docs/api/rabbitmq.com.ref.asciidoc [hunk 1]
```diff
@@ -84,7 +84,7 @@ PersistentVolumeClaim is an embedded version of k8s.io/api/core/v1.PersistentVol
 [id="{anchor_prefix}-github-com-rabbitmq-cluster-operator-api-v1beta1-plugin"]
 ==== Plugin (string) 
 
-
+kubebuilder validating tags 'Pattern' and 'MaxLength' must be specified on string type. Alias type 'string' as 'Plugin' to specify schema validation on items of the list 'AdditionalPlugins'
```
Comment (Zerpet) [inline docs/api/rabbitmq.com.ref.asciidoc:88]: This comment doesn't make any sense. It looks like an error message from kubebuilder or `code-gen`. We should remove this (not include it for commit) or investigate and fix the root cause.

**Review by Zerpet — CHANGES_REQUESTED** [review]
We might need @coro's input to address the comment below.

File: docs/api/rabbitmq.com.ref.asciidoc (hunk 1 above)
Comment (ChunyiLyu) [reply docs/api/rabbitmq.com.ref.asciidoc]: removed

**Review by Zerpet — APPROVED** [review]

**Pull request merged**